### PR TITLE
correct ipv6 address handling when building MongoClients

### PR DESCRIPTION
**Pull request by evanchooly**
fixes JIRA-2245

### CODE REVIEW

File: driver-core/src/main/com/mongodb/ServerAddress.java [hunk 1]
```diff
@@ -129,6 +129,16 @@ public ServerAddress(final String host, final int port) {
         this.port = portToUse;
     }
 
+    private String join(final String delimiter, final String[] array, final int start, final int end) {
```
Comment (rozza) [inline driver-core/src/main/com/mongodb/ServerAddress.java]: Don't think this method is needed.

**Comment by rozza** [comment]
Other than the join method - LGTM

File: driver-core/src/main/com/mongodb/ServerAddress.java (hunk 1 above)
Comment (evanchooly) [inline driver-core/src/main/com/mongodb/ServerAddress.java]: You're right.  Was using that for a failed approach and missed it in the clean up.

**Comment by jyemin** [comment]
lgtm

**Pull request merged**